### DISCUSSION TITLE
nextbox-ui-plugin

### DISCUSSION THREAD

**Discussion by Izanatos** [netbox-community/netbox-docker#821] [Getting Help]
Hi Guys,

very new working with docker and netbox but I managed to get portainer and netbox running.

I tried to install nextbox-ui-plugin via the console access of the netbox docker container but i am getting : 
**ERROR: Could not install packages due to an OSError: [Errno 13] Permission denied: '/opt/netbox/venv/lib/python3.10/site-packages/netbox_topology_views'
Check the permissions.**

Any advise as how I could get this plugin running ?

thanks,

**Comment by PaulR282** [comment]
You need to install the package when creating the image. Check the Dockerfile of the nextbox repo. With this you can build an image with the plugin installed. You then need to add it in the `configuration.py`
I modified the Dockerfile, here is mine:

```
ARG FROM_IMAGE=docker.io/netboxcommunity/netbox
ARG FROM_TAG=v3.2-2.1.0
ARG FROM=${FROM_IMAGE}:${FROM_TAG}
FROM ${FROM}

ENV VIRTUAL_ENV=/opt/netbox/venv
ENV PATH="$VIRTUAL_ENV/bin:$PATH"
COPY ./nextbox_ui_plugin /source/nextbox-ui-plugin/nextbox_ui_plugin/
COPY ./setup.py /source/nextbox-ui-plugin/
COPY ./MANIFEST.in /source/nextbox-ui-plugin/
COPY ./README.md /source/nextbox-ui-plugin/
COPY --chown=0:0 --chmod=755 ./nextbox_ui_plugin/static/nextbox_ui_plugin /opt/netbox/netbox/static/nextbox_ui_plugin
RUN pip3 install --no-cache-dir nextbox-ui-plugin
RUN pip3 install dnspython #thas for a report
```
